storage/storage_image.go

```diff
@@ -0,0 +1,430 @@
+package storage
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/image"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/containers/storage/storage"
+)
+
+var (
+	// ErrBlobDigestMismatch is returned when PutBlob() is given a blob
+	// with a digest-based name that doesn't match its contents.
+	ErrBlobDigestMismatch = errors.New("blob digest mismatch")
+)
+
+type storageImage struct {
+	store          storage.Store
+	imageRef       *storageReference
+	Tag            string            `json:"tag,omitifempty"`
```

Comment:
It should be, though in practice these haven't tended to be empty, so I hadn't yet noticed.  Fixing.  Thanks!